rust/src/measurements/make_private_lazyframe/aggregate/mod.rs

```diff
@@ -0,0 +1,189 @@
+        PartitionDistance(input_metric.clone()),
+        StabilityMap::new_fallible(move |&d_in| {
+            let l0 = margin.max_influenced_partitions.unwrap_or(d_in).min(d_in);
+            let li = margin.max_partition_contributions.unwrap_or(d_in).min(d_in);
+            let l1 = l0.inf_mul(&li)?.min(d_in);
+            Ok((l0, l1, li))
+        }),
+    )?;
+
+    let m_exprs = (t_group_by
+        >> make_basic_composition(
+            aggs.into_iter()
+                .map(|expr| {
+                    make_private_expr(
+                        expr_domain.clone(),
+                        PartitionDistance(input_metric.clone()),
+                        output_measure.clone(),
+                        expr.clone(),
+                        param,
+                    )
+                })
+                .collect::<Fallible<_>>()?,
+        )?)?;
+
+    let f_exprs = m_exprs.function.clone();
+    let privacy_map = m_exprs.privacy_map.clone();
+
+    t_prior
+        >> Measurement::new(
+            input_domain,
+            Function::new_fallible(move |arg: &LogicalPlan| {
+                let mut output = plan.clone();
+                if let LogicalPlan::Aggregate {
+                    ref mut input,
+                    ref mut aggs,
+                    ..
+                } = output
+                {
+                    *input = Box::new(arg.clone());
+                    *aggs = f_exprs.eval(&(arg.clone(), all()))?;
+                };
+                Ok(output)
+            }),
+            input_metric,
+            output_measure,
+            privacy_map,
+        )?
+}
+
+#[cfg(test)]
+mod test_make_group_by {
+    use crate::domains::{AtomDomain, Margin, SeriesDomain};
+    use crate::error::ErrorVariant::MakeMeasurement;
+    use crate::error::*;
+    use crate::measures::MaxDivergence;
+    use polars::prelude::*;
+
+    use crate::metrics::SymmetricDistance;
+
+    use super::*;
+
+    #[test]
+    fn test_aggregate() -> Fallible<()> {
+        let lf_domain = LogicalPlanDomain::new(vec![
+            SeriesDomain::new("A", AtomDomain::<i32>::default()),
+            SeriesDomain::new("B", AtomDomain::<f64>::default()),
+            SeriesDomain::new("C", AtomDomain::<i32>::default()),
+        ])?
+        .with_margin(&["A", "C"], Margin::new().with_public_keys())?;
+
+        let lf = df!(
+            "A" => &[1i32, 2, 2],
+            "B" => &[1.0f64, 2.0, 2.0],
+            "C" => &[8i32, 9, 10],)?
+        .lazy();
+
+        let error_variant_res = make_private_aggregate::<_, SymmetricDistance, _>(
+            lf_domain,
+            SymmetricDistance,
+            MaxDivergence::<f64>::default(),
+            1.,
+            lf.clone()
+                .group_by(&[col("A"), col("C")])
+                .agg(&[col("B").sum()])
+                .logical_plan,
+        )
+        .map(|_| ())
+        .unwrap_err()
+        .variant;
+
+        assert_eq!(MakeMeasurement, error_variant_res);
```

Comment:
If I'm reading this right, this tests that the error is the error we expected. But a test of successful behavior would also make sense, I'd think?

Reply:
It would, but I don't actually have a way to parse the expressions yet.

Reply:
It would, but I don't actually have a way to turn the expressions into measurements yet, at this point of the PR stack.